// @generated/type-graphql/resolvers/outputs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MovieSumAggregate = exports.MovieMinAggregate = exports.MovieMaxAggregate = exports.MovieGroupBy = exports.MovieCountAggregate = exports.MovieAvgAggregate = exports.AggregateMovie = exports.AffectedRowsOutput = void 0;
var AffectedRowsOutput_1 = require("./AffectedRowsOutput");
Object.defineProperty(exports, "AffectedRowsOutput", { enumerable: true, get: function () { return AffectedRowsOutput_1.AffectedRowsOutput; } });
var AggregateMovie_1 = require("./AggregateMovie");
Object.defineProperty(exports, "AggregateMovie", { enumerable: true, get: function () { return AggregateMovie_1.AggregateMovie; } });
var MovieAvgAggregate_1 = require("./MovieAvgAggregate");
Object.defineProperty(exports, "MovieAvgAggregate", { enumerable: true, get: function () { return MovieAvgAggregate_1.MovieAvgAggregate; } });
var MovieCountAggregate_1 = require("./MovieCountAggregate");
Object.defineProperty(exports, "MovieCountAggregate", { enumerable: true, get: function () { return MovieCountAggregate_1.MovieCountAggregate; } });
var MovieGroupBy_1 = require("./MovieGroupBy");
Object.defineProperty(exports, "MovieGroupBy", { enumerable: true, get: function () { return MovieGroupBy_1.MovieGroupBy; } });
var MovieMaxAggregate_1 = require("./MovieMaxAggregate");
Object.defineProperty(exports, "MovieMaxAggregate", { enumerable: true, get: function () { return MovieMaxAggregate_1.MovieMaxAggregate; } });
var MovieMinAggregate_1 = require("./MovieMinAggregate");
Object.defineProperty(exports, "MovieMinAggregate", { enumerable: true, get: function () { return MovieMinAggregate_1.MovieMinAggregate; } });
var MovieSumAggregate_1 = require("./MovieSumAggregate");
Object.defineProperty(exports, "MovieSumAggregate", { enumerable: true, get: function () { return MovieSumAggregate_1.MovieSumAggregate; } });
